### ISSUE TITLE
random crash

### ISSUE THREAD

**Issue by trollmad3** [Wurst-Imperium/Wurst-Client#1084]
--------------- Wurst Client Crash Report ----------------
---- Please report this crash at: wurst-client.tk/bugs ----
// I blame Apple.

Time: 6/27/15 7:37 PM
Description: Unexpected error

java.util.ConcurrentModificationException
	at java.util.ArrayList$Itr.checkForComodification(Unknown Source)
	at java.util.ArrayList$Itr.next(Unknown Source)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:543)
	at com.google.common.collect.TransformedIterator.hasNext(TransformedIterator.java:43)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:542)
	at net.minecraft.util.ChatComponentStyle.getFormattedText(ChatComponentStyle.java:107)
	at net.minecraft.client.gui.GuiNewChat.drawChat(GuiNewChat.java:97)
	at net.minecraft.client.gui.GuiIngame.func_175180_a(GuiIngame.java:414)
	at net.minecraft.client.renderer.EntityRenderer.updateCameraAndRender(EntityRenderer.java:1176)
	at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1327)
	at net.minecraft.client.Minecraft.run(Minecraft.java:412)
	at net.minecraft.client.main.Main.main(Main.java:127)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Stacktrace:
	at java.util.ArrayList$Itr.checkForComodification(Unknown Source)
	at java.util.ArrayList$Itr.next(Unknown Source)
	at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:543)
	at com.google.common.collect.TransformedIterator.hasNext(TransformedIterator.java:43)
	at com.google.common.collect.Iterators$5.hasNext(Iterators.java:542)
	at net.minecraft.util.ChatComponentStyle.getFormattedText(ChatComponentStyle.java:107)
	at net.minecraft.client.gui.GuiNewChat.drawChat(GuiNewChat.java:97)
	at net.minecraft.client.gui.GuiIngame.func_175180_a(GuiIngame.java:414)

-- Affected level --
Details:
	Level name: MpServer
	All players: 6 total; [EntityPlayerSP['bugzncheese1'/420545, l='MpServer', x=-111.09, y=10.48, z=-55.00], EntityOtherPlayerMP['SGTsela'/150, l='MpServer', x=-112.70, y=10.26, z=-53.48], EntityOtherPlayerMP['mine223girl'/102, l='MpServer', x=-99.50, y=-175.94, z=-93.50], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-111.69, y=9.00, z=-53.69], EntityOtherPlayerMP['epicrooster'/43, l='MpServer', x=-113.06, y=13.09, z=-101.31], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-70.44, y=14.26, z=-102.20]]
	Chunk stats: MultiplayerChunkCache: 441, 441
	Level seed: 0
	Level generator: ID 00 - default, ver 1. Features enabled: false
	Level generator options: 
	Level spawn location: -76.00,74.00,-82.00 - World: (-76,74,-82), Chunk: (at 4,4,14 in -5,-6; contains blocks -80,0,-96 to -65,255,-81), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Level time: 19620 game time, 73123 day time
	Level dimension: 0
	Level storage version: 0x00000 - Unknown?
	Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
	Level game mode: Game mode: adventure (ID 2). Hardcore: false. Cheats: false
	Forced entities: 47 total; [EntityPlayerSP['bugzncheese1'/420545, l='MpServer', x=-111.09, y=10.48, z=-55.00], EntityOtherPlayerMP['mine223girl'/102, l='MpServer', x=-100.04, y=11.50, z=-75.22], EntityOtherPlayerMP['ImmortalFrieza'/229, l='MpServer', x=-99.50, y=-219.80, z=-94.53], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-111.69, y=9.00, z=-53.69], EntityOtherPlayerMP['SGTsela'/150, l='MpServer', x=-112.70, y=10.26, z=-53.48], EntityArrow['arrow'/9161, l='MpServer', x=-73.00, y=9.25, z=-107.94], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-111.69, y=9.00, z=-53.69], EntityOtherPlayerMP['Soukie'/207, l='MpServer', x=-46.54, y=29.45, z=-84.32], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-64.63, y=14.00, z=-104.41], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-103.50, y=-207.50, z=-91.06], EntityOtherPlayerMP['Graystripe4588'/73, l='MpServer', x=-43.85, y=35.01, z=-72.01], EntityOtherPlayerMP['epicrooster'/43, l='MpServer', x=-107.53, y=13.00, z=-64.31], EntityOtherPlayerMP['Rowandude66'/147, l='MpServer', x=-109.31, y=12.00, z=-117.66], EntityArrow['arrow'/9744, l='MpServer', x=-107.34, y=14.38, z=-64.97], EntityOtherPlayerMP['scottsdavison'/34, l='MpServer', x=-46.91, y=29.68, z=-86.75], EntityOtherPlayerMP['Icrazedragon'/42, l='MpServer', x=-112.31, y=13.00, z=-130.47], EntityOtherPlayerMP['Graystripe4588'/73, l='MpServer', x=-46.46, y=34.16, z=-76.78], EntityOtherPlayerMP['scottsdavison'/34, l='MpServer', x=-80.84, y=15.38, z=-103.50], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-70.44, y=14.26, z=-102.20], EntityOtherPlayerMP['mine223girl'/102, l='MpServer', x=-99.50, y=-175.94, z=-93.50], EntityOtherPlayerMP['RABUFA'/181, l='MpServer', x=-96.47, y=11.56, z=-80.44], EntityOtherPlayerMP['SGTsela'/150, l='MpServer', x=-112.70, y=10.26, z=-53.48], EntityOtherPlayerMP['RABUFA'/181, l='MpServer', x=-106.31, y=10.28, z=-51.79], EntityOtherPlayerMP['mkmo5_X'/55, l='MpServer', x=-116.50, y=12.00, z=-118.50], EntityOtherPlayerMP['Rowandude66'/147, l='MpServer', x=-80.42, y=11.19, z=-94.98], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-124.84, y=11.69, z=-119.94], EntityOtherPlayerMP['ImmortalFrieza'/229, l='MpServer', x=-103.53, y=1.00, z=-91.38], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-113.22, y=9.00, z=-50.31], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-71.78, y=10.00, z=-106.31], EntityOtherPlayerMP['gerrhard123'/74, l='MpServer', x=-110.94, y=12.41, z=-115.72], EntityOtherPlayerMP['PancakeKingdom10'/117, l='MpServer', x=-116.50, y=12.00, z=-118.50], EntityOtherPlayerMP['SGTsela'/150, l='MpServer', x=-126.26, y=0.94, z=-81.96], EntityOtherPlayerMP['PancakeKingdom10'/117, l='MpServer', x=-65.02, y=14.80, z=-102.95], EntityOtherPlayerMP['gerrhard123'/74, l='MpServer', x=-73.56, y=10.91, z=-103.72], EntityOtherPlayerMP['Rowandude66'/147, l='MpServer', x=-130.63, y=12.25, z=-107.40], EntityArrow['arrow'/8418, l='MpServer', x=-63.66, y=14.03, z=-106.31], EntityArrow['arrow'/10213, l='MpServer', x=-45.63, y=30.02, z=-86.94], EntityOtherPlayerMP['RABUFA'/181, l='MpServer', x=-111.06, y=12.00, z=-120.09], EntityOtherPlayerMP['ImmortalFrieza'/229, l='MpServer', x=-95.50, y=11.00, z=-147.50], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-74.43, y=7.05, z=-105.13], EntityOtherPlayerMP['mine223girl'/102, l='MpServer', x=-99.50, y=-175.94, z=-93.50], EntityOtherPlayerMP['epicrooster'/43, l='MpServer', x=-113.06, y=13.09, z=-101.31], EntityArrow['arrow'/10091, l='MpServer', x=-113.00, y=10.05, z=-53.47], EntityOtherPlayerMP['mkmo5_X'/55, l='MpServer', x=-111.56, y=12.00, z=-117.75], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-70.44, y=14.26, z=-102.20], EntityOtherPlayerMP['epicrooster'/43, l='MpServer', x=-113.06, y=13.09, z=-101.31], EntityArrow['arrow'/10427, l='MpServer', x=-47.83, y=36.60, z=-64.93]]
	Retry entities: 0 total; []
	Server brand: BeeCord (1.8) <- Spigot
	Server type: Non-integrated multiplayer server
Stacktrace:
	at net.minecraft.client.multiplayer.WorldClient.addWorldInfoToCrashReport(WorldClient.java:393)
	at net.minecraft.client.Minecraft.addGraphicsAndWorldToCrashReport(Minecraft.java:2954)
	at net.minecraft.client.Minecraft.run(Minecraft.java:441)
	at net.minecraft.client.main.Main.main(Main.java:127)

-- System Details --
Details:
	Minecraft Version: 1.8
	Wurst Client Version: 1.15 (latest: 1.15)
	Operating System: Windows 7 (x86) version 6.1
	Java Version: 1.8.0_45, Oracle Corporation
	Java VM Version: Java HotSpot(TM) Client VM (mixed mode), Oracle Corporation
	Memory: 181229600 bytes (172 MB) / 382984192 bytes (365 MB) up to 523501568 bytes (499 MB)
	JVM Flags: 6 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx512M -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
	IntCache: cache: 0, tcache: 62, allocated: 13, tallocated: 95
	Launched Version: Wurst
	LWJGL: 2.9.1
	OpenGL: GeForce GTX 750 Ti/PCIe/SSE2 GL version 4.5.0 NVIDIA 350.12, NVIDIA Corporation
	GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

	Using VBOs: No
	Is Modded: Very likely; Jar signature invalidated
	Type: Client (map_client.txt)
	Resource Packs: [faithful32pack.zip]
	Current Language: Canadian English (Canada)
	Profiler Position: N/A (disabled)

**Comment by mcccs** [comment]
```
--------------- Wurst Client Crash Report ----------------
---- Please report this crash at: wurst-client.tk/bugs ----
// I blame Apple.

Time: 6/27/15 7:37 PM
Description: Unexpected error

java.util.ConcurrentModificationException
at java.util.ArrayList$Itr.checkForComodification(Unknown Source)
at java.util.ArrayList$Itr.next(Unknown Source)
at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
at com.google.common.collect.Iterators$5.hasNext(Iterators.java:543)
at com.google.common.collect.TransformedIterator.hasNext(TransformedIterator.java:43)
at com.google.common.collect.Iterators$5.hasNext(Iterators.java:542)
at net.minecraft.util.ChatComponentStyle.getFormattedText(ChatComponentStyle.java:107)
at net.minecraft.client.gui.GuiNewChat.drawChat(GuiNewChat.java:97)
at net.minecraft.client.gui.GuiIngame.func_175180_a(GuiIngame.java:414)
at net.minecraft.client.renderer.EntityRenderer.updateCameraAndRender(EntityRenderer.java:1176)
at net.minecraft.client.Minecraft.runGameLoop(Minecraft.java:1327)
at net.minecraft.client.Minecraft.run(Minecraft.java:412)
at net.minecraft.client.main.Main.main(Main.java:127)

A detailed walkthrough of the error, its code path and all known details is as follows:

-- Head --
Stacktrace:
at java.util.ArrayList$Itr.checkForComodification(Unknown Source)
at java.util.ArrayList$Itr.next(Unknown Source)
at com.google.common.collect.TransformedIterator.next(TransformedIterator.java:48)
at com.google.common.collect.Iterators$5.hasNext(Iterators.java:543)
at com.google.common.collect.TransformedIterator.hasNext(TransformedIterator.java:43)
at com.google.common.collect.Iterators$5.hasNext(Iterators.java:542)
at net.minecraft.util.ChatComponentStyle.getFormattedText(ChatComponentStyle.java:107)
at net.minecraft.client.gui.GuiNewChat.drawChat(GuiNewChat.java:97)
at net.minecraft.client.gui.GuiIngame.func_175180_a(GuiIngame.java:414)

-- Affected level --
Details:
Level name: MpServer
All players: 6 total; [EntityPlayerSP['bugzncheese1'/420545, l='MpServer', x=-111.09, y=10.48, z=-55.00], EntityOtherPlayerMP['SGTsela'/150, l='MpServer', x=-112.70, y=10.26, z=-53.48], EntityOtherPlayerMP['mine223girl'/102, l='MpServer', x=-99.50, y=-175.94, z=-93.50], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-111.69, y=9.00, z=-53.69], EntityOtherPlayerMP['epicrooster'/43, l='MpServer', x=-113.06, y=13.09, z=-101.31], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-70.44, y=14.26, z=-102.20]]
Chunk stats: MultiplayerChunkCache: 441, 441
Level seed: 0
Level generator: ID 00 - default, ver 1. Features enabled: false
Level generator options: 
Level spawn location: -76.00,74.00,-82.00 - World: (-76,74,-82), Chunk: (at 4,4,14 in -5,-6; contains blocks -80,0,-96 to -65,255,-81), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
Level time: 19620 game time, 73123 day time
Level dimension: 0
Level storage version: 0x00000 - Unknown?
Level weather: Rain time: 0 (now: false), thunder time: 0 (now: false)
Level game mode: Game mode: adventure (ID 2). Hardcore: false. Cheats: false
Forced entities: 47 total; [EntityPlayerSP['bugzncheese1'/420545, l='MpServer', x=-111.09, y=10.48, z=-55.00], EntityOtherPlayerMP['mine223girl'/102, l='MpServer', x=-100.04, y=11.50, z=-75.22], EntityOtherPlayerMP['ImmortalFrieza'/229, l='MpServer', x=-99.50, y=-219.80, z=-94.53], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-111.69, y=9.00, z=-53.69], EntityOtherPlayerMP['SGTsela'/150, l='MpServer', x=-112.70, y=10.26, z=-53.48], EntityArrow['arrow'/9161, l='MpServer', x=-73.00, y=9.25, z=-107.94], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-111.69, y=9.00, z=-53.69], EntityOtherPlayerMP['Soukie'/207, l='MpServer', x=-46.54, y=29.45, z=-84.32], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-64.63, y=14.00, z=-104.41], EntityOtherPlayerMP['J_block11'/72, l='MpServer', x=-103.50, y=-207.50, z=-91.06], EntityOtherPlayerMP['Graystripe4588'/73, l='MpServer', x=-43.85, y=35.01, z=-72.01], EntityOtherPlayerMP['epicrooster'/43, l='MpServer', x=-107.53, y=13.00, z=-64.31], EntityOtherPlayerMP['Rowandude66'/147, l='MpServer', x=-109.31, y=12.00, z=-117.66], EntityArrow['arrow'/9744, l='MpServer', x=-107.34, y=14.38, z=-64.97], EntityOtherPlayerMP['scottsdavison'/34, l='MpServer', x=-46.91, y=29.68, z=-86.75], EntityOtherPlayerMP['Icrazedragon'/42, l='MpServer', x=-112.31, y=13.00, z=-130.47], EntityOtherPlayerMP['Graystripe4588'/73, l='MpServer', x=-46.46, y=34.16, z=-76.78], EntityOtherPlayerMP['scottsdavison'/34, l='MpServer', x=-80.84, y=15.38, z=-103.50], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-70.44, y=14.26, z=-102.20], EntityOtherPlayerMP['mine223girl'/102, l='MpServer', x=-99.50, y=-175.94, z=-93.50], EntityOtherPlayerMP['RABUFA'/181, l='MpServer', x=-96.47, y=11.56, z=-80.44], EntityOtherPlayerMP['SGTsela'/150, l='MpServer', x=-112.70, y=10.26, z=-53.48], EntityOtherPlayerMP['RABUFA'/181, l='MpServer', x=-106.31, y=10.28, z=-51.79], EntityOtherPlayerMP['mkmo5_X'/55, l='MpServer', x=-116.50, y=12.00, z=-118.50], EntityOtherPlayerMP['Rowandude66'/147, l='MpServer', x=-80.42, y=11.19, z=-94.98], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-124.84, y=11.69, z=-119.94], EntityOtherPlayerMP['ImmortalFrieza'/229, l='MpServer', x=-103.53, y=1.00, z=-91.38], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-113.22, y=9.00, z=-50.31], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-71.78, y=10.00, z=-106.31], EntityOtherPlayerMP['gerrhard123'/74, l='MpServer', x=-110.94, y=12.41, z=-115.72], EntityOtherPlayerMP['PancakeKingdom10'/117, l='MpServer', x=-116.50, y=12.00, z=-118.50], EntityOtherPlayerMP['SGTsela'/150, l='MpServer', x=-126.26, y=0.94, z=-81.96], EntityOtherPlayerMP['PancakeKingdom10'/117, l='MpServer', x=-65.02, y=14.80, z=-102.95], EntityOtherPlayerMP['gerrhard123'/74, l='MpServer', x=-73.56, y=10.91, z=-103.72], EntityOtherPlayerMP['Rowandude66'/147, l='MpServer', x=-130.63, y=12.25, z=-107.40], EntityArrow['arrow'/8418, l='MpServer', x=-63.66, y=14.03, z=-106.31], EntityArrow['arrow'/10213, l='MpServer', x=-45.63, y=30.02, z=-86.94], EntityOtherPlayerMP['RABUFA'/181, l='MpServer', x=-111.06, y=12.00, z=-120.09], EntityOtherPlayerMP['ImmortalFrieza'/229, l='MpServer', x=-95.50, y=11.00, z=-147.50], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-74.43, y=7.05, z=-105.13], EntityOtherPlayerMP['mine223girl'/102, l='MpServer', x=-99.50, y=-175.94, z=-93.50], EntityOtherPlayerMP['epicrooster'/43, l='MpServer', x=-113.06, y=13.09, z=-101.31], EntityArrow['arrow'/10091, l='MpServer', x=-113.00, y=10.05, z=-53.47], EntityOtherPlayerMP['mkmo5_X'/55, l='MpServer', x=-111.56, y=12.00, z=-117.75], EntityOtherPlayerMP['MaxPvPx'/148, l='MpServer', x=-70.44, y=14.26, z=-102.20], EntityOtherPlayerMP['epicrooster'/43, l='MpServer', x=-113.06, y=13.09, z=-101.31], EntityArrow['arrow'/10427, l='MpServer', x=-47.83, y=36.60, z=-64.93]]
Retry entities: 0 total; []
Server brand: BeeCord (1.8) <- Spigot
Server type: Non-integrated multiplayer server
Stacktrace:
at net.minecraft.client.multiplayer.WorldClient.addWorldInfoToCrashReport(WorldClient.java:393)
at net.minecraft.client.Minecraft.addGraphicsAndWorldToCrashReport(Minecraft.java:2954)
at net.minecraft.client.Minecraft.run(Minecraft.java:441)
at net.minecraft.client.main.Main.main(Main.java:127)

-- System Details --
Details:
Minecraft Version: 1.8
Wurst Client Version: 1.15 (latest: 1.15)
Operating System: Windows 7 (x86) version 6.1
Java Version: 1.8.0_45, Oracle Corporation
Java VM Version: Java HotSpot(TM) Client VM (mixed mode), Oracle Corporation
Memory: 181229600 bytes (172 MB) / 382984192 bytes (365 MB) up to 523501568 bytes (499 MB)
JVM Flags: 6 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xmx512M -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn128M
IntCache: cache: 0, tcache: 62, allocated: 13, tallocated: 95
Launched Version: Wurst
LWJGL: 2.9.1
OpenGL: GeForce GTX 750 Ti/PCIe/SSE2 GL version 4.5.0 NVIDIA 350.12, NVIDIA Corporation
GL Caps: Using GL 1.3 multitexturing.
Using GL 1.3 texture combiners.
Using framebuffer objects because OpenGL 3.0 is supported and separate blending is supported.
Shaders are available because OpenGL 2.1 is supported.
VBOs are available because OpenGL 1.5 is supported.

Using VBOs: No
Is Modded: Very likely; Jar signature invalidated
Type: Client (map_client.txt)
Resource Packs: [faithful32pack.zip]
Current Language: Canadian English (Canada)
Profiler Position: N/A (disabled)
```

**Comment by GummiBoat** [comment]
# *Duplicate: #642*

![](https://camo.githubusercontent.com/fde0e4b68d36e2d72331263e4c421c56965a71df/687474703a2f2f692e696d6775722e636f6d2f714a4a514930582e706e67)
